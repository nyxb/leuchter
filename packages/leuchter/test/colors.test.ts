import { expect, test } from 'vitest'
import { getThemeColors } from '..'

test('get theme colors from theme name', async () => {
   const result = await getThemeColors('github-dark')
   expect(result).toMatchSnapshot()
})

test('get theme colors from theme name from-css', async () => {
   const result = await getThemeColors('github-from-css')
   expect(result).toMatchSnapshot()
})

test('get theme colors from theme undefined', async () => {
   expect(getThemeColors(undefined)).rejects.toThrow()
})

test('get theme colors from unknown theme', async () => {
   expect(getThemeColors('unknown-theme' as any)).rejects.toThrow()
})

test('get theme colors from empty theme', async () => {
   const result = await getThemeColors({})
   expect(result).toMatchSnapshot()
})
