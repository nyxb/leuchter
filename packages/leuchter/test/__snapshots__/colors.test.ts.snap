// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`get theme colors from theme name from-css 1`] = `
{
  "background": "var(--ch-16)",
  "colorScheme": "var(--ch-0)",
  "editor": {
    "background": "var(--ch-16)",
    "foreground": "var(--ch-4)",
    "infoForeground": "var(--ch-19)",
    "lineHighlightBackground": "var(--ch-17)",
    "rangeHighlightBackground": "var(--ch-18)",
    "selectionBackground": "var(--ch-20)",
  },
  "editorGroup": {
    "border": "var(--ch-23)",
  },
  "editorGroupHeader": {
    "tabsBackground": "var(--ch-22)",
  },
  "editorLineNumber": {
    "foreground": "var(--ch-24)",
  },
  "focusBorder": "var(--ch-21)",
  "foreground": "var(--ch-4)",
  "icon": {
    "foreground": "var(--ch-15)",
  },
  "input": {
    "background": "var(--ch-16)",
    "border": "var(--ch-23)",
    "foreground": "var(--ch-4)",
  },
  "lighter": {
    "inlineBackground": "var(--ch-26)",
  },
  "list": {
    "activeSelectionBackground": "var(--ch-25)",
    "activeSelectionForeground": "var(--ch-4)",
    "hoverBackground": "var(--ch-17)",
    "hoverForeground": "var(--ch-4)",
  },
  "sideBar": {
    "background": "var(--ch-22)",
    "border": "var(--ch-23)",
    "foreground": "var(--ch-4)",
  },
  "tab": {
    "activeBackground": "var(--ch-16)",
    "activeBorder": "var(--ch-16)",
    "activeForeground": "var(--ch-4)",
    "border": "var(--ch-23)",
    "inactiveBackground": "var(--ch-22)",
    "inactiveForeground": "var(--ch-15)",
  },
}
`;
